wp/plugin/Makefile

```diff
@@ -1,28 +1,55 @@
-PASS_NAME = wp
+PASS_NAME := wp
+SHELL=/bin/bash
 
 SAMPLE_BIN_DIR = ../resources/sample_binaries
 API_PATH = $(shell bap --api-list-paths)
 
-.PHONY: all build install clean verifier
+BUILD := _build/build_unique_id
+SRC_FILES = $(wildcard */*.ml) $(wildcard */*.mli)
 
-all: install
+.PHONY: all verifier test uninstall uninstall.verifier
 
-build:
-	bapbuild -use-ocamlfind -pkgs 'z3,bap_wp,re' -tag 'warn(A-48-44)' -I lib $(PASS_NAME).plugin
+all: install
 
-install: build verifier
+install: $(BUILD) | verifier
 	bapbundle update -desc 'Computes the weakest precondition of a subroutine given a postcondition.' $(PASS_NAME).plugin
 	bapbundle install $(PASS_NAME).plugin
 
-test: install
-	ocamlbuild -r -use-ocamlfind -pkgs 'bap,z3,bap_wp,oUnit,re' -tag 'warn(A-48-44),debug,thread' -Is "lib,tests" test.native
+build: $(BUILD)
+
+$(BUILD): $(SRC_FILES)
+# exactly one version of bap_wp installed
+ifneq ($(shell opam pin list | grep "bap_wp" | wc -l), 1)
+	$(error "bap_wp lib is not installed. Please install using relevant makefile.")
+endif
+	printf "SO YA MESSED WITH: " $(SRC_FILES)
```

Comment:
This makes sense! It felt like a hack because I was included a temporary file used exclusively to figure out if the project needed to be rebuilt. You're right--we can use `wp.plugin` as the file to watch (and the name of the target). This makes things feel much better.

I added the error message to let the user know what went wrong. However, this is redundant; I'll remove it.

asdjfk I missed deleting that print statement. Was using it to debug changed files haha